### PR TITLE
Add android support showcase

### PR DESCRIPTION
**Pull request by gengjiawen**
Fix https://github.com/libuv/help/issues/145

### CODE REVIEW

File: libuv-demo/src/main/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        -DCMAKE_LIBRARY_OUTPUT_DIRECTORY=${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
+        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+        -DANDROID_NDK=${ANDROID_NDK}
+        -DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}
+        -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}
+        -DCMAKE_MAKE_PROGRAM=${CMAKE_MAKE_PROGRAM}
+        -DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}
+        -G${CMAKE_GENERATOR})
+
+ExternalProject_Add(libuv
```
Comment (gengjiawen) [inline libuv-demo/src/main/cpp/CMakeLists.txt]: Looks like first time not auto download this ? What do I missing ? @bnoordhuis

Comment (gengjiawen) [reply libuv-demo/src/main/cpp/CMakeLists.txt]: For now, I have to remove ${libuv_static} from last line, and libuv is download, add it back.

Comment (gengjiawen) [reply libuv-demo/src/main/cpp/CMakeLists.txt]: Turns out this is related to ninja.

**Comment by gengjiawen** [comment]
@bnoordhuis

**Comment by gengjiawen** [comment]
ping @bnoordhuis cc @cjihrig

**Comment by cjihrig** [comment]
I don't quite understand the vision for this repo enough to sign off on this. If this repo is just supposed to include a collection of libuv-adjacent projects, then it might be a good idea to split them up in separate directories (ie this pull request would add a top level directory, and everything else would be inside of that). It would be good to get feedback from @bnoordhuis and @saghul here.

**Comment by bnoordhuis** [comment]
Yes, I agree. It's supposed to be a collection of examples/showcases/etc. so moving everything from this PR into its own top-level directory is the right thing to do. LGTM apart from that.

**Comment by gengjiawen** [comment]
I have moved all the files to `android` folder.

**Comment by gengjiawen** [comment]
> I have moved all the files to `android` folder.

@bnoordhuis

**Comment by gengjiawen** [comment]
ping @bnoordhuis

**Comment by bnoordhuis** [comment]
Thanks for the ping, Jiawen. Merged in 58b3cb6, cheers. :-)

**Pull request closed without merging**